### PR TITLE
lapack/gonum: add Dpstf2 and its test

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

Updates #1717

### CODE REVIEW

File: lapack/gonum/dpstf2.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		return true, 0
+	}
+
+	switch {
+	case len(a) < (n-1)*lda+n:
+		panic(shortA)
+	case len(piv) != n:
+		panic(badLenPiv)
+	case len(work) < 2*n:
+		panic(shortWork)
```
Comment (kortschak) [inline lapack/gonum/dpstf2.go:47]: Comment for work length requirement?

File: lapack/gonum/dpstf2.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+// The factorization has the form
+//  Pᵀ * A * P = Uᵀ * U ,  if uplo = blas.Upper,
+//  Pᵀ * A * P = L  * Lᵀ,  if uplo = blas.Lower,
+// where U is an upper triangular matrix and L is lower triangular, and P is
+// stored as vector piv.
+//
+// Dpstf2 does not attempt to check that A is positive semidefinite.
+//
+// Dpstf2 is an internal routine. It is exported for testing purposes.
+func (Implementation) Dpstf2(uplo blas.Uplo, n int, a []float64, lda int, piv []int, tol float64, work []float64) (ok bool, rank int) {
```
Comment (kortschak) [inline lapack/gonum/dpstf2.go:26]: The order of the return values is surprising, all our others are `(some interestingvalue T, ok bool)`.

**Review by soypat — COMMENTED** [review]
Went over `lapack/gonum/dpstf2.go`. Can't say much for the tests.

File: lapack/gonum/dpstf2.go (hunk 1 above)
Comment (vladimir-ch) [reply lapack/gonum/dpstf2.go:50]: Done, and also for `piv` to be exactly `n`.

**Review by kortschak — APPROVED** [review]

**Pull request merged**